data-prepper-api/src/main/java/org/opensearch/dataprepper/logging/SensitiveArgumentMaskingConverter.java

```diff
@@ -0,0 +1,45 @@
+package org.opensearch.dataprepper.logging;
+
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.pattern.ConverterKeys;
+import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.apache.logging.log4j.core.pattern.PatternConverter;
+import org.apache.logging.log4j.message.Message;
+import org.slf4j.helpers.MessageFormatter;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.EVENT;
+import static org.opensearch.dataprepper.logging.DataPrepperMarkers.SENSITIVE;
+
+@Plugin(name = "sensitiveArgumentMaskingConverter", category = PatternConverter.CATEGORY)
+@ConverterKeys({"mask"})
+public class SensitiveArgumentMaskingConverter extends LogEventPatternConverter {
```

Comment:
This should go into `data-prepper-core`. It is a feature of Data Prepper and not part of the API.